### PR TITLE
feat: description for apps and jobs

### PR DESCRIPTION
**Pull request by ShashwatDadhich**
# Description

Description added for apps and jobs same as that of cluster

Fixes # [Issue](https://github.com/devtron-labs/dashboard/issues/1159)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] It has been tested locally
- [x] It has been tested by deploying the code to VM

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/types.ts [hunk 1]
```diff
@@ -78,13 +79,20 @@ export interface AppMetaInfo {
     appId: number
     appName: string
     createdBy: string
-    description: string
+    description: description
```
Comment (shivani170) [inline src/components/app/types.ts:82]: rename it to Description

Comment (ShashwatDadhich) [reply src/components/app/types.ts]: Done

File: src/components/app/create/CreateApp.tsx [hunk 2]
```diff
@@ -332,20 +349,41 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
...
+                    <>
+                        <span className="form__label mt-16">Description</span>
+                        {this.state.showClusterDescription ? (
+                            <div className="h-auto">
+                                <ClusterDescription
+                                    isClusterTerminal={false}
+                                    isSuperAdmin={true}
+                                    appId={this.state.form.appId}
+                                    descriptionId={0}
+                                    initialDescriptionText={''}
```
Comment (shivani170) [inline src/components/app/create/CreateApp.tsx:361]: Can we avoid passing these empty props ?

Comment (ShashwatDadhich) [reply src/components/app/create/CreateApp.tsx]: done

**Review by shivani170 — APPROVED** [review]

File: src/components/app/list-new/Constants.ts [hunk 3]
```diff
@@ -29,6 +29,15 @@ export const APPLIST_EMPTY_STATE_MESSAGING = {
     noAppsFoundInfoText: `We couldn't find any matching applications.`,
 }
 export const ClearFiltersLabel = 'Clear filters'
+export const DefaultAppNote =`## Describe this application
+
```
Comment (vivek-devtron) [inline src/components/app/list-new/Constants.ts:33]: We should use some other option for formatting adding new line like this is not the correct way

File: src/components/common/Description/GenericDescription.scss [hunk 4]
```diff
@@ -0,0 +1,140 @@
+.tab-list {
```
Comment (vivek-devtron) [inline src/components/common/Description/GenericDescription.scss:1]: this should be wrapped in parent container class

File: src/components/app/create/CreateApp.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/app/create/CreateApp.tsx:359]: Not yet done

File: src/components/common/Description/GenericDescription.tsx [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+                                        type="button"
+                                        onClick={toggleDescriptionView}
+                                    >
+                                        Cancel
+                                    </button>
+                                    <button
+                                        data-testid="description-edit-save-button"
+                                        className="cta flex h-36"
+                                        type="submit"
+                                        onClick={() => {
```
Comment (vivek-devtron) [inline src/components/common/Description/GenericDescription.tsx:433]: extract out this method

File: src/components/app/list-new/Constants.ts (hunk 3 above)
Comment (ShashwatDadhich) [reply src/components/app/list-new/Constants.ts:33]: done

File: src/components/common/Description/GenericDescription.tsx (hunk 5 above)
Comment (ShashwatDadhich) [reply src/components/common/Description/GenericDescription.tsx]: done

File: src/components/charts/discoverChartDetail/DiscoverChartDetails.tsx [hunk 6]
```diff
@@ -384,9 +384,10 @@ function isReadmeInputCheckbox(text: string) {
     }
     return false;
 }
-export function MarkDown({ markdown = '', className = '', breaks = false, ...props }) {
+export function MarkDown({ markdown = '', className = '', breaks = false, disableEscapedText = false, setExpandableIcon = undefined, ...props }) {
```
[the remark this answers is not in the document]
Comment (ShashwatDadhich) [reply src/components/charts/discoverChartDetail/DiscoverChartDetails.tsx:387]: done

File: src/components/app/create/CreateApp.tsx (hunk 2 above)
Comment (ShashwatDadhich) [reply src/components/app/create/CreateApp.tsx]: done now

File: src/components/common/Description/GenericDescription.scss (hunk 4 above)
Comment (ShashwatDadhich) [reply src/components/common/Description/GenericDescription.scss]: done

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**